Python/bltinmodule.c

```diff
@@ -2667,15 +2736,22 @@ builtin_sum_impl(PyObject *module, PyObject *iterable, PyObject *start)
                 int overflow;
                 value = PyLong_AsLongAndOverflow(item, &overflow);
                 if (!overflow) {
-                    f_result += (double)value;
+                    cr_result.sum += (double)value;
+                    ci_result.sum += 0.0;
                     Py_DECREF(item);
                     continue;
                 }
             }
-            if (c && isfinite(c)) {
-                f_result += c;
+            if (PyFloat_Check(item)) {
+                double value = PyFloat_AS_DOUBLE(item);
+                cr_result.sum += value;
+                ci_result.sum += 0.0;
+                Py_DECREF(item);
```

Comment:
I think you need might need to swap this with `_Py_DECREF_SPECIALIZED(item, _PyFloat_ExactDealloc);` above.